libbeat/processors/condition.go

```diff
@@ -257,16 +267,32 @@ func (c *Condition) checkEquals(event ValuesMap) bool {
 			if intValue != equalValue.Int {
 				return false
 			}
-		} else {
-			sValue, err := extractString(value)
-			if err != nil {
-				logp.Warn("unexpected type %T in equals condition as it accepts only integers and strings. ", value)
+
+			continue
+		}
+
+		sValue, err := extractString(value)
+
+		if err == nil {
+			if sValue != equalValue.Str {
 				return false
 			}
-			if sValue != equalValue.Str {
+
+			continue
+		}
+
+		bValue, err := extractBool(value)
+
+		if err == nil {
+			if bValue != equalValue.Bool {
 				return false
 			}
+
+			continue
 		}
+
+		logp.Warn("unexpected type %T in equals condition as it accepts only integers, strings or bools. ", value)
```

Comment:
Can we change this to `logp.Err`? In general we prefer not to use `Warn` and make things either `Info` or `Err`.

Reply:
Done